### PR TITLE
Port "Transfer funds" transaction type section

### LINKED ISSUE #737: Port "Transfer funds" transaction type section
Clarifications:
- On click, the max button should populate the field with the available balance on the Safe of the selected token.
- The token selector should only display the tokens that the safe currently has.
- Both the tokens of the safe and the balance of said token can be obtained from the Safe Transaction Sevice API: https://safe-transaction-mainnet.safe.global/

Image

Image

### PR DESCRIPTION
**Pull request by thomassdk**
## Description

Ports over the functionality of transfer funds section without validation.

To test add the safes from those defined in `Looking for safes and modules to test the dialogs?` and play around

[Screencast from 2023-07-21 10-45-18.webm](https://github.com/JoinColony/colonyCDapp/assets/32598350/39e59885-2580-4feb-8a20-170c5769d503)

**Changes** 🏗

* As much as possible, the types have been converted to use the new types defined by the CDapp. For this reason, the data received from the safe balances API is typed and then converted into the CDapp's `Token` type

Resolves #737

### CODE REVIEW

File: src/components/common/Dialogs/ControlSafeDialog/AmountBalances/AmountBalances.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+interface Props {
+  safeBalances: SafeBalance[];
+  disabledInput: boolean;
+  selectedSafe: Safe | undefined;
+  transactionFormIndex: number;
+  maxButtonParams: MaxButtonParams;
+  /* handleChange: () => void; */
+}
+
+const displayName = 'common.ControlSafeDialog.AmountBalances';
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/ControlSafeDialog/AmountBalances/AmountBalances.tsx:31]: I think this AmountBalances is used only inside `ControlSafeForm`. Therefore the id should be `common.ControlSafeDialog.ControlSafeForm.AmountBalances`

File: src/components/common/Dialogs/ControlSafeDialog/AmountBalances/AmountTokens.css [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  left: -6px;
+}
+
+.networkFee {
+  position: relative;
+  bottom: 14px;
+  font-size: var(--size-tiny);
+  font-weight: var(--weight-bold);
+  text-align: right;
+  color: var(--temp-grey-blue-7);
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/ControlSafeDialog/AmountBalances/AmountTokens.css:50]: I think this color no longer exists and it was replaced by `--grey-purple`

File: src/components/common/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferFundsSection.tsx [hunk 3]
```diff
@@ -1,14 +1,239 @@
...
+        <AmountBalances
+          selectedSafe={selectedSafe}
+          safeBalances={safeBalances}
+          disabledInput={disabledInput}
+          transactionFormIndex={transactionFormIndex}
+          /* handleValidation={handleValidation} */
+          /* handleChange={handleInputChange} */
+          maxButtonParams={{
+            maxAmount: `${formattedSafeBalance}`,
+            options: {},
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferFundsSection.tsx:194]: I don't think we need this `options`, right?

**Review by ArmandoGraterol — COMMENTED** [review]
Now that the multi-transaction stuff exists, you should be able to rebase on top of it and add whatever piece was missing before it :)

File: src/components/common/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferFundsSection.tsx (hunk 3 above)
Comment (thomassdk) [reply src/components/common/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferFundsSection.tsx:194]: on further inspection ive removed all the placeholders for manual validation checks and replaced them with these two options in the button
```ts
            options: { shouldTouch: true, shouldValidate: true },
```

**Review by ArmandoGraterol — COMMENTED** [review]
Great work! Before I approve, I would like to get the safe balance validation included:
FireShot Capture 135 - Actions - Colony - armando - qa-xdai colony io

Right now I can go beyond the safe's balance without getting any errors.

File: src/components/common/Dialogs/ControlSafeDialog/AmountBalances/AmountBalances.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+     *   return (
+     *     safeNetworkData?.name.toLowerCase() ||
+     *     ETHEREUM_NETWORK.name.toLowerCase()
+     *   );
+     * }; */
+
+    return {
+      ...(safeNetworkData?.nativeToken || ETHEREUM_NETWORK.nativeToken)!,
+      tokenAddress: ADDRESS_ZERO,
+      /* networkName: getNetworkName(), */
```
[the remark this answers is not in the document]
Comment (thomassdk) [reply src/components/common/Dialogs/ControlSafeDialog/AmountBalances/AmountBalances.tsx:78]: Hmm so I left this in as a talking point as I am unsure of what we are using the networkName for. Is this something that was used in the Dapp but is no longer needed in the CDapp?

**Comment by thomassdk** [comment]
Thanks for the review @ArmandoGraterol! 

I've been leaving things in as I was unsure of their usage in the new CDapp, so it's great to get your eyes on it.

Everything should be fixed now and the validation working correctly

[Screencast from 2023-08-01 15-02-13.webm](https://github.com/JoinColony/colonyCDapp/assets/32598350/e50b9977-0221-4994-addc-8ed05e581131)

File: src/components/common/Dialogs/ControlSafeDialog/AmountBalances/AmountBalances.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/components/common/Dialogs/ControlSafeDialog/AmountBalances/AmountBalances.tsx:78]: Hmmm okay so from a quick search in the Dapp, I think this prop was used before to get the token icon?
```
        const networkName = 'networkName' in token ? token.networkName : '';
        const response = await loadTokenImages(address, networkName);
        if (response.ok) {
          const blob = await response.blob();
          const base64image = await getBase64image(blob);
```

But I think this is no longer the case for the CDapp as far as I can see, so we should be good if we remove this code.

**Review by ArmandoGraterol — APPROVED** [review]
Nice work my friend 🎉 LGTM

**Pull request merged**